app/controllers/UploadController.scala

```diff
@@ -89,9 +91,9 @@ class UploadController(override val authActions: HMACAuthActions, awsConfig: AWS
       video
   }
 
-  private def getPart(id: String, key: String): Option[UploadPart] = for {
-    upload <- stepFunctions.getById(id)
-    part <- upload.parts.find(_.key == key)
+  private def getPart(credentialsRequest: UploadCredentialsRequest): Option[UploadPart] = for {
+    upload <- stepFunctions.getById(credentialsRequest.atomId)
```

Comment:
turns out this is the wrong ID 😢  step function invocations are identified by `s"$atomId-$version"`. Closing this PR for now, will reopen another once I've got it working.